src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -160,6 +160,22 @@ internal Conversions Conversions
         /// <summary>Lazily caches SyntaxTrees by their xxHash128 checksum. Used to look up the syntax tree referenced by an interceptor.</summary>
         private ImmutableSegmentedDictionary<ReadOnlyMemory<byte>, OneOrMany<SyntaxTree>> _contentHashToSyntaxTree;
 
+        internal ExtendedErrorTypeSymbol ImplicitlyTypedVariableUsedInForbiddenZoneType
+        {
+            get
+            {
+                return field ??= new ExtendedErrorTypeSymbol(this, name: "var", arity: 0, errorInfo: null, variableUsedBeforeDeclaration: true);
```

Comment:
Is there any risk of race condition (two threads concurrently see a `null` field and decide to set, but to different values, which would mess up downstream phases which check for singleton value)?